paddle/fluid/framework/ir/mkldnn/cpu_quantize_squash_pass.cc

```diff
@@ -152,8 +152,13 @@ void CPUQuantizeSquashPass::DequantQuantSquash(
     GET_IR_NODE_FROM_SUBGRAPH(next_op, next_op, squash_pattern);
 
     // Don't squash if e.g. just one concat input is unsigned
-    if (IsDequantizeInputUint8(dequant_in) &&
-        !quant_op->Op()->GetAttrIfExists<bool>("is_negative_input")) {
+    bool is_concat_signed =
+        quant_op->Op()->GetAttrIfExists<bool>("is_negative_input");
+    bool is_input_unsigned = IsDequantizeInputUint8(dequant_in);
+    if (is_concat_signed && is_input_unsigned) {
+      VLOG(4) << "Do not squash dequant-quant, because "
+              << "is_concat_signed: " << is_concat_signed
+              << ", is_input_unsigned: " << is_input_unsigned << ".";
```

Comment:
Please add a check if the next operator is `concat` or `elementwise`, because only these operators have several inputs and they all have to have the same data type.
For elementwise check, you can add a comment to remove it when the BinaryMkldnn kernel will support two different data types.

Reply:
done